diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/KdocMethods.kt

```diff
@@ -66,18 +76,40 @@ class KdocMethods : Rule("kdoc-methods") {
         isFixMode = autoCorrect
         emitWarn = emit
 
-        if (node.elementType == FUN && node.getFirstChildWithType(MODIFIER_LIST).isAccessibleOutside()) {
-            val config = KdocMethodsConfiguration(configRules.getRuleConfig(MISSING_KDOC_ON_FUNCTION)?.configuration
-                    ?: mapOf())
-            val isTestMethod = node.hasTestAnnotation() || node.isLocatedInTest(params.fileName!!.splitPathToDirs(), config.testAnchors)
-            if (!isTestMethod && !node.isStandardMethod() && !node.isSingleLineGetterOrSetter()) {
-                checkSignatureDescription(node)
+        if (node.elementType == FUN) {
+            if (node.hasChildOfType(KDOC)){
+                checkKDocParam(node)
```

Comment:
It's better to do this check inside `checkSignatureDescription`, where comparison of declared and documented parameters is already being made. You can split it into smaller methods if you think it is becoming unreadable.